### DISCUSSION TITLE
[Help]: How to disable scrollNext

### DISCUSSION THREAD

**Discussion by Yuschyk** [davidjerleke/embla-carousel#1001] [Help] [answered]
### Summary

I need to disable switching to the next slide on a certain condition (until the user does something on the current slide). How can I implement this, as I didn't see this setting in the carousel.

<img width="478" alt="image" src="https://github.com/user-attachments/assets/53285dd9-6fb9-440f-be3b-dea9f30abffc">

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment]
Hi @Yuschyk,

Thanks for your question? Is the carousel in question draggable?

**Reply by Yuschyk** [reply]
Hello @davidjerleke,

Thank you for your answer. Yes, my carousel is draggable. And i want to disable drag events to next slide when user doesn't complete different conditions on current slide

**Reply by Yuschyk** [reply]
@davidjerleke It looks like this:

https://github.com/user-attachments/assets/66fdb1a7-b4d7-43a1-8154-68a8b773bad6

**Reply by davidjerleke** [reply]
@Yuschyk do you want to temporarily disable drag interactions in both directions?

**Reply by Yuschyk** [reply]
@davidjerleke Yes, in should be like `allowSlideNext` in swiper

**Reply by davidjerleke** [reply]
@Yuschyk in that case you can do this: 

```tsx
import useEmblaCarousel from 'embla-carousel-react'

export function EmblaCarousel() {
  const allowDragRef = useRef(true) // Set this ref to true if you want the carousel to be draggable, and false when you want to disable it

  const [emblaRef] = useEmblaCarousel({
    watchDrag: () => {
      return allowDragRef.current
    }
  })

  // ...
}
```

Docs reference for [watchDrag](https://www.embla-carousel.com/api/options/#watchdrag).

**Reply by Yuschyk** [reply]
@davidjerleke Yes, but your solution has a little probrem. watchDrag disable not only next slide. When watchDrag user can't drag to previous slide. And in this case, we only need to lock the next slide.

**Reply by davidjerleke** [reply, marked as answer]
> Yes, but your solution has a little probrem. watchDrag disable not only next slide. When watchDrag user can't drag to previous slide. And in this case, we only need to lock the next slide.

Yes, that's why I asked this question: 

> @Yuschyk do you want to temporarily disable drag interactions in both directions?

And you answered **yes**, but then you actually meant **no** I guess 🙂, because you only want to disable drag to the next slide. However, this is not doable at the time of writing. You'll have to wait for the upcoming feature `watchDragMove` before you can achieve this.

This discussion is a duplicate of:

- #749
